Gifski/Gifski.swift

```diff
@@ -2,31 +2,34 @@ import Foundation
 import AVFoundation
 
 final class Gifski {
-	private var frameCount = 0
-	private var frameIndex = 0
 	private(set) var isRunning = false
-	private(set) var progress: Double = 0
+	private var progress: Progress!
+
+	// `progress.fractionCompleted` is KVO-compliant, but we expose this for convenience
 	var onProgress: ((_ progress: Double) -> Void)?
 
 	/**
 	- parameters:
 		- frameRate: Clamped to 5...30. Uses the frame rate of `inputUrl` if not specified.
 	*/
+	@discardableResult
 	func convertFile(
 		_ inputUrl: URL,
 		outputUrl: URL,
 		quality: Double = 1,
 		dimensions: CGSize? = nil,
 		frameRate: Int? = nil
-	) {
+	) -> Progress {
+		/// TODO: Find a better way to handle this
 		guard !isRunning else {
-			return
+			fatalError("Create a new instance if you want to run multiple conversions at once")
 		}
 
-		frameCount = 0
-		frameIndex = 0
 		isRunning = true
 
+		progress = Progress(parent: nil, userInfo: [.fileURLKey: outputUrl])
```

Comment:
I think it's better to pass `Progress.current()` as parent here, so that it can be observed elsewhere with progress.becomeCurrent...

Reply:
Any reason you set UserInfo and fileURL?

Reply:
It's for https://github.com/sindresorhus/gifski-app/pull/9#discussion_r170493129. I'm not sure if both are needed and what the difference is though.